rundeckapp/grails-spa/packages/ui-trellis/src/app/components/home/services/homeServices.ts

```diff
@@ -25,5 +25,57 @@ export async function getSummary(): Promise<any> {
         // e.message in this case is the error message from the server response
         throw {message: "Error: " + e.message, response: e.response};
     }
+}
 
+export async function getProjectNames(): Promise<any> {
+    let appLinks = getAppLinks();
+    try {
+        const response = await axios
+            .get(appLinks.menuProjectNamesAjax, {
```

Comment:
let's avoid using the old "ajax" endpoints, we should only use API endpoints. for this purpose, maybe the endpoint to use is the project list, but specify no metadata params, to get the list without metadata.

Reply:
Yes, I'm aware that shouldn't be using the old ajax endpoints but this endpoint had the same issue of the activity where it is relying on the session to populate the data, so didn't try to convert it:

```
def projectNamesAjax() {
        AuthContext authContext = rundeckAuthContextProcessor.getAuthContextForSubject(session.subject)
        def fprojects = frameworkService.refreshSessionProjects(authContext, session)

        render(contentType:'application/json',text:
                ([projectNames: fprojects] )as JSON
        )
    }
```

If it's fine to ignore the session logic, can make a new endpoint that just returns the project names but I'm suspecting that the session logic is exactly what is causing the endpoint to return data fast and that blindly converting might backfire. 

But I think there's a confusion here - the service call getProjectNames (that uses the URL  menuProjectNamesAjax) isn't fetching the metadata, it's the other endpoint (getProjects) that does it, and it's only fetching the necessary one (authz, config and message)